### PR TITLE
drivers/note: remove choice in Kconfig

### PR DESCRIPTION
**Pull request by Gary-Hobson**
## Summary
remove choice in Kconfig， to support multiple note drivers to be used at the same time

### CODE REVIEW

File: drivers/note/Kconfig [hunk 1]
```diff
@@ -226,21 +208,32 @@ config DRIVER_NOTERAM
...
+
+endif
 
 config DRIVER_NOTELOG
 	bool "Note syslog driver"
-	select SCHED_INSTRUMENTATION_EXTERNAL
 	---help---
 		The note driver output to syslog.
 
 config SEGGER_SYSVIEW
```
Comment (xiaoxiang781216) [inline drivers/note/Kconfig:234]: move to segger/Kconfig

File: drivers/note/Kconfig [hunk 2]
```diff
@@ -195,12 +180,9 @@ config DRIVER_NOTECTL
 		If this option is selected, the instrumentation filter control device
 		/dev/notectl is provided.
 
-choice
-	prompt "Note driver selection"
-	default DRIVER_NOTERAM
-
 config DRIVER_NOTERAM
 	bool "Note RAM driver"
+	default y
```
Comment (xiaoxiang781216) [inline drivers/note/Kconfig:185]: let's keep the default off

Comment (Gary-Hobson) [reply drivers/note/Kconfig:185]: DRIVER_NOTE is off by default
We should enable at least one note driver after enabling DRIVER_NOTE

File: drivers/note/Kconfig (hunk 1 above)
Comment (Gary-Hobson) [reply drivers/note/Kconfig]: done

File: drivers/note/Kconfig (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/note/Kconfig:185]: Ok

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**